### ISSUE TITLE
The location setter should prevent cycles in the location graph

### ISSUE THREAD

**Issue by rlazarus** [rlazarus/MUSS#97]
Not only should putting A in A be prohibited, putting A in B should be prohibited if B is in A, or if B is in C and C is in A, and so on.

... Or should it? That would be bizarre, but I think it wouldn't break anything, and it could be used to interesting effect. Tagging as "discuss" rather than "bug" for now.

**Comment by duress-to-kill** [comment]
On the subject of twilight zone spatial mechanics, is it a well-founded assumption that an Object can only be in one place at a time?
Allowing a thing to be in multiple places might solve the utility problems with a cyclically-nested object, or set of objects. Maybe somebody needs to put a moebius strip in their pocket?
Obviously this would have the potential to cause really confusing accidental behavior for the unsuspecting user. If this were to be allowed, it might be a good idea to have "bilocating" an object require using a special command for that purpose.

**Comment by relsqui** [comment]
I have the same feeling about this as I did in my previous comment, but here's a simpler version: I would need to hear a plausible and constructive reason this _should_ exist before I thought it was worth our time to implement. Yes, we can provide unrealistic abstractions--this is a tool for building fantastical worlds, after all--but the whole point of the engine is to provide some ready-made structure and boundaries that matches what its users expect and want. Sure, we _can_ provide arbitrary made-up levels of abstraction, but at that point, why have an engine?

TL;DR I'm agin' it unless you can convince me that builders using the engine will expect and want to have it available.

**Comment by duress-to-kill** [comment]
I don't feel strongly about it. I'm fine leaving it out.
It was my thought that by restricting that capacity to a different command (which would likely rarely get used by most player and therefore be more esoteric) the structure provided by the engine would still be there, and that it would probably be minimal effort to code. but I'll grant, I don't feel motivated to write it in.
